subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/SyncTaskIntegrationTest.groovy

```diff
@@ -394,36 +393,86 @@ class SyncTaskIntegrationTest extends AbstractIntegrationSpec {
         ins.close()
     }
 
-    @ToBeImplemented
     @Issue("https://github.com/gradle/gradle/issues/9586")
-    @Requires(TestPrecondition.CASE_INSENSITIVE_FS)
-    def "change in case of input file still syncs properly"() {
+    def "change in case of input files and folders will still syncs properly"() {
```

Comment:
* change file case only without content
* break it up into smaller once (single file / directory rename)

Reply:
Fixed